src/Xrefcheck/Util/Interpolate.hs

```diff
@@ -0,0 +1,54 @@
+{- SPDX-FileCopyrightText: 2018-2019 Serokell <https://serokell.io>
+ -
+ - SPDX-License-Identifier: MPL-2.0
+ -}
+{-# OPTIONS_GHC -Wno-unrecognised-pragmas #-}
+
+module Xrefcheck.Util.Interpolate where
+
+import Universum
+
+import Data.Text.Lazy qualified as TL
+import Data.Text.Lazy.Builder (fromLazyText, toLazyText)
+import Fmt (Buildable, Builder, blockListF, blockListF', indentF)
+
+{-# HLINT ignore "Avoid functions that generate extra trailing newlines/whitespaces" #-}
+
```

Comment:
Let's add some docs to explain why we needed to write these functions in the first place. It'll help future readers understand the context.

```suggestion

{- $notes

The `blockListF` and `indentF` frunctions from @fmt@ add a trailing newline, which makes them unsuitable for string interpolation.

Consider this case:

> [int||
> aaa
> #{indentF 2 "bbb"}
> ccc
> |]

One would reasonably expect this to produce:

> aaa
>   bbb
> ccc

But, in reality, it produces:

> aaa
>   bbb
>
> ccc

This module introduces versions of these functions that do not produce a trailing newline
and can therefore be safely used in string interpolation.
-}

```